browser/components/EditProject.tsx

```diff
@@ -0,0 +1,315 @@
+/* Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import * as React from 'react';
+import SweetAlert from 'react-bootstrap-sweetalert';
+import Select from 'react-select';
+import { postJSON, reqJSON } from '../util/index';
+
+interface Props {
+  groupList: [
+    {
+      goal: string;
+      group_id: number;
+      group_name: string;
+      contribMTD: number;
+      contribMonth: number;
+      contribWeek: number;
+      contribYear: number;
+      numGroups: number;
+      numUsers: number;
+      projects: number[];
+      sponsor: string;
+    }
+  ];
+  project: string;
+}
+
+interface State {
+  alert: any;
+  groupList: [
+    {
+      goal: string;
+      group_id: number;
+      group_name: string;
+      contribMTD: number;
+      contribMonth: number;
+      contribWeek: number;
+      contribYear: number;
+      numGroups: number;
+      numUsers: number;
+      projects: number[];
+      sponsor: string;
+    }
+  ];
+  projectGroups: string;
+  projectID: number;
+  projectInfo?: {
+    groups: [
+      {
+        contribMTD: number;
+        contribMonth: number;
+        contribWeek: number;
+        contribYear: number;
+        goal: string;
+        group_id: number;
+        group_name: string;
+        projects: number[];
+        sponsor: string;
+      }
+    ];
+    project: {
+      project_auto_aprovable: boolean;
+      project_id: number;
+      project_license: string;
+      project_name: string;
+      project_url: string;
+      project_verified: boolean;
+    };
+    users: [
+      {
+        company_alias: string;
+        contribMTD: number;
+        contribMonth: number;
+        contribWeek: number;
+        contribYear: number;
+        github_alias: string;
+      }
+    ];
+  };
+  projectLicense: string;
+  projectName: string;
+  projectURL: string;
+  projectUsers: string;
+}
+
+export default class EditProject extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alert: null,
+      projectGroups: '',
+      projectID: null,
+      projectInfo: null,
+      projectLicense: '',
+      projectName: '',
+      projectURL: '',
+      projectUsers: '',
+      groupList: null,
+    };
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
```

Comment:
This may not do what you think it does -- it's like doing a pointer comparison. The contents aren't being compared; only whether they're the same reference. This may either spam out updates or fail to update when you think it should, depending on React's mood at any given moment.

Reply:
I was trying something and missed this. Removing.